### ISSUE TITLE
PyPortal.get_local_time() calls incorrect IP/domain

### ISSUE THREAD

**Issue by eloekset** [adafruit/Adafruit_CircuitPython_PortalBase#86]
I'm using PyPortal 7.X to create a weather station display for Netatmo. I have an on-prem ASP.NET 6 proxy against the Netatmo API, so PyPortal qeries my proxy, which queries Netatmo and prepares the response in a json structure that's easy to display as widgets on the PyPortal.

I have the following program loop:
```
while True:
    # only query the online time once per hour (and on first run)
    if (not localtile_refresh) or (time.monotonic() - localtile_refresh) > 3600:
        try:
            print("Getting time from internet!")
            pyportal.get_local_time()
            localtile_refresh = time.monotonic()
        except RuntimeError as e:
            print("Some error occured, retrying! -", e)
            continue

    # only query the weather every 10 minutes (and on first run)
    if (not weather_refresh) or (time.monotonic() - weather_refresh) > 600:
        try:
            value = pyportal.fetch()
            gfx.draw_display(value)
            weather_refresh = time.monotonic()
        except RuntimeError as re:
            print("Some error occured, retrying! -", re)
            gfx.draw_error()
            continue
        except TimeoutError as te:
            print("Timeout - ", te)
            gfx.draw_error()
            continue

    gfx.draw_time()
    updateTime = .5
    if mode == "weekday":
        updateTime = 30

    time.sleep(updateTime)  # wait X seconds before updating anything again
```

I experience unexpected behavior when `pyportal.get_local_time()` executes the second time (and all times after). First time the Adafruit API at http://io.adafruit.com is called as expected. But the second time, even though the URL looks correct, my local ASP.NET proxy is called instead. That's the same IP as `pyportal.fetch()` calls.

This is a flowchart describing what happens:
![image](https://user-images.githubusercontent.com/791721/211209717-bbdbeca5-7b84-4c78-803f-1560286694e9.png)

I added Azure Application Insights to my ASP.NET proxy to figure out what's going on, and was surprised to see requests for the domain `io.adafruit.com` hitting my on-prem proxy. I've documented proof of the traffic in my own project in this issue:
https://github.com/blog-eivindgl-com/netatmo-pyportal-display/issues/9

I don't know why this happens, but I suspect there's some kind of local DNS issue on the PyPortal that incorrectly maps `io.adafruit.com` to the same IP address as the URL of the `fetch()` call.

**Comment by makermelissa** [comment]
Interesting. This will need further investigation. It doesn't look like io.adafruit.com is redirecting, so it should be hitting the same url each time.

**Comment by eloekset** [comment]
I've run the PyPortal display in my own apartment for quite a while now, and the issue doesn't occur here. I guess the router in my mother's house was causing the issue.
![image](https://user-images.githubusercontent.com/791721/211998348-09b3e345-f470-444a-8f73-fe5f07db66f3.png)

**Comment by eloekset** [comment]
No, I'm sorry. This is not a router's fault. It happened again on my network this night. It has happened 16 times since last time PyPortal crashed, and I haven't done anything but pressing the reset button. It's hooked up a charger, so it's not connected to my computer. I don't know what's going on here, but it's obvious that suddenly traffic to `http://io.adafruit.com` gets routed to my service running on `http://192.168.1.104`. (I use a different computer to host my Docker container now, but it's still hitting that same on-prem Netatmo Proxy shown in the diagram above.)

![image](https://user-images.githubusercontent.com/791721/212463037-4953f2b7-57d7-4f83-8d51-bdc8bd5b170d.png)

I've now fixed the timezone issue on my implementation of the API, so I'm actually thinking about modifying the PyPortal source code to query my own implementation instead of `http://io.adafruit.com` when `PyPortal.get_local_time()` is executed. Since my implementation is compatible and returns the correct value now, it should just work fine:
![image](https://user-images.githubusercontent.com/791721/212463153-41311ad8-571b-4ec7-a135-a82110ddc381.png)

The reason I'm considering that, is that PyPortal crashes quite often when refreshing the local time. Most often there's a **TimeoutError**:
![image](https://user-images.githubusercontent.com/791721/212463216-cd4267c1-13c6-49ef-aac0-0a0e5f6b74d1.png)

Line 48 is where this function is called:
![image](https://user-images.githubusercontent.com/791721/212463335-bf71503d-39fa-404d-a31a-1ee77663a89f.png)

I don't know if it will fix the **TimeoutError** issue, but it's worth trying.

But this issue where traffic is suddenly hitting the wrong server should be investigated further. It could still be a router issue, but since this now has happened on two different LANs, I think it's more likely something happing on the PyPortal device - maybe in the **adafruit_requests.py** or the **adafruit_portalbase/network.py**?

**Comment by makermelissa** [comment]
For the timeout error, there's already an issue open in the library where the error is occurring. You can follow it here: https://github.com/adafruit/Adafruit_CircuitPython_ESP32SPI/issues/170
